ext/nokogiri_ext_xmlsec/nokogiri_sign.c

```diff
@@ -187,6 +208,20 @@ VALUE sign(VALUE self, VALUE rb_opts) {
     exception_message = "signature failed";
     goto done;
   }
+  if (store_references) {
+    rb_references = rb_ary_new_capa((long)xmlSecPtrListGetSize(&dsigCtx->signedInfoReferences));
```

Comment:
This cast should be checked for bounds since we're going from size_t -> long and size_t might be larger than long.